appsec-kit-v24/src/main/java/com/vaadin/appsec/v24/views/AppSecView.java

```diff
@@ -37,7 +37,7 @@
 @Push
 @PreserveOnRefresh
```

Comment:
I would also remove this, it probably creates more side-effects than providing value for the use cases of the kit.

Reply:
I removed the `@PreserveOnRefresh` annotation. The `@Push` annotation is needed for the `Scan now` button to work properly. When the scan is finished an event is sent to the client and based on that the `Scan now` button will be enabled again. Without `@Push` it doesn't work.